firmware/src/widgets/stockwidget/StockWidget.cpp

```diff
@@ -46,11 +56,8 @@ void StockWidget::draw(bool force) {
 void StockWidget::update(bool force) {
     if (force || m_stockDelayPrev == 0 || (millis() - m_stockDelayPrev) >= m_stockDelay) {
         setBusy(true);
-        for (int8_t i = 0; i < m_stockCount; i++) {
-            getStockData(m_stocks[i]);
-        }
-        setBusy(false);
-        m_stockDelayPrev = millis();
+        // Create a task to handle all stock updates
+        xTaskCreate(taskGetStockData, "StockDataTask", 8192, this, 1, NULL);
```

Comment:
How do you ensure that this task is not created again while it is running?

Reply:
good point...maybe need to make use of the TaskHandle_t in FreeRTOS to check first if its running.  I'll try some testing for that